### DISCUSSION TITLE
SSR issues with Handsontable

### DISCUSSION THREAD

**Discussion by jessapp** [remix-run/remix#1604] [Q&A] [answered]
# Problem

I am importing [Handsontable](https://handsontable.com/) to my Remix application . When I create an instance of `new Handsontable`, I get SSR errors. See [this repository](https://github.com/jessapp/hot-remix-repro) for a reproduction (go to `app/routes/table.tsx` to see the Handsontable instance). 

![Screen Shot 2022-01-21 at 1 58 04 PM](https://user-images.githubusercontent.com/18542356/150606115-cdf7b48e-b4ee-4fcd-9f0e-4eae5148a045.png)

# Solutions I have tried

As per [this discussion thread](https://github.com/remix-run/remix/discussions/1023), I tried renaming the impacted file from `app/routes/table.tsx` to `app/routes/table.client.tsx` to prevent the code from being used in the server, but the error persisted. 

I also tried creating a ClientOnly component, as per [this GitHub issue](https://github.com/remix-run/remix/issues/180), but it also didn't solve the issue. 

Is there any way to import Handsontable without triggering SSR issues? Or to disable SSR altogether? Thank you!

**Comment by sergiodxa** [comment, marked as answer]
> I tried renaming the impacted file from `app/routes/table.tsx` to `app/routes/table.client.tsx`

You can't have .client files inside the routes folder, try re-exporting that Handsontable package from a .client file and then importing inside a route.

**Reply by jessapp** [reply]
Hey @sergiodxa, thanks for your reply! 

Importing Handsontable from a .client file outside of the routes folder did indeed get me past the error I posted above, however the instance of Handsontable never rendered on the page. 

When I combined your solution with the ClientOnly solution I mentioned in my initial question, I was able to get past the SSR error and render the Handsontable instance.

Thanks for your help!
